<p>
Classes in <code>com.sun.*</code> and <code>sun.*</code> packages are considered implementation details, and are not part of the Java API.
They can cause problems when moving to new versions of Java because there is no backwards compatibility guarantee.
Such classes are almost always wrapped by Java API classes that should be used instead.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
import com.sun.jna.Native;     // Non-Compliant
import sun.misc.BASE64Encoder; // Non-Compliant
</pre>