Having a class and some of its methods sharing the same name is misleading, and leaves others to wonder whether it was done that way on purpose, or was the methods supposed to be a constructor.

<h2>Noncompliant Code Example</h2>
<pre>
class Foo {
  public Foo() {...}
  public int Foo(String label) {...} // Noncompliant
}
</pre>

<h2>Compliant Solution</h2>
<pre>
class Foo {
  public Foo() {...}
  public int foo(String label) {...} // Compliant
}
</pre>
