<p>
  Instantiating a StringBuilder or a StringBuffer with a character is misleading because most Java developers would expect the character to be the initial value of the StringBuffer.
</p>
<p>
  What actually happens is that the int representation of the character is used to determine the initial size of the StringBuffer.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
StringBuffer foo = new StringBuffer('x');   //equivalent to StringBuffer foo = new StringBuffer(120);
</pre>

<h2>Compliant Solution</h2>

<pre>
StringBuffer foo = new StringBuffer("x");
</pre>
