<p>There are two possible syntaxes for a lambda having only one input parameter with an inferred type: with and without parentheses around that single parameter. The simpler syntax, without parentheses, is more compact and readable than the one with parentheses, and is therefore preferred.</p>

<h2>Noncompliant Code Example</h2>
<pre>
(x) -> x * 2
</pre>

<h2>Compliant Solution</h2>
<pre>
x -> x * 2
</pre>
